vignettes/Data_Normalization.Rmd

```diff
@@ -0,0 +1,384 @@
+
+We can use the `min_prop` argument to the `normalize_global` function to enforce a minimum proportion of biomolecules required to apply a normalization. In the following code using the LOS (0.1) subset, we would get an error if we included `min_prop = 0.2` because the proportion of biomolecules in the specified subset is below that threshold.
+
+```{r}
+# LOS MAD
+mynorm <- normalize_global(omicsData = mypep, 
+                           subset_fn = "los", 
+                           params = list(los = (0.1)),
+                           norm_fn = "mad", 
+                           apply_norm = FALSE)
+mynorm$prop_features_calc
+```
+
+Here we demonstrate how to use the PPP subset and z-score normalization function.
+
+```{r}
+# PPP z-score
+mynorm <- normalize_global(omicsData = mypep, 
+                           subset_fn = "ppp", 
+                           params = list(ppp = (0.5)),
+                           norm_fn = "zscore", 
+                           apply_norm = FALSE,
+                           min_prop = 0.2)
+mynorm$prop_features_calc
+```
+
+Here we demonstrate how to use the PPP-RIP subset and specify parameter values, along with the median normalization function.
+
+```{r}
+# PPP-RIP median
+mynorm <- normalize_global(omicsData = mypep, 
+                           subset_fn = "ppp_rip", 
+                           params = list(ppp = (0.5), rip = (0.2)),
+                           norm_fn = "median", 
+                           apply_norm = FALSE)
+mynorm$prop_features_calc
+```
+
+Finally, we demonstrate how to use the set of complete biomolecules with the mean function.
+
+```{r}
+# Complete mean
+mynorm <- normalize_global(omicsData = mypep, 
+                           subset_fn = "complete", 
+                           norm_fn = "mean", 
+                           apply_norm = FALSE)
+mynorm$prop_features_calc
+```
+
+
+For proteomics data, or large lipidomics or metabolomics datasets (>200ish biomolecules??), the Statistical Procedure for the Analyses of peptide abundance Normalization Strategies (SPANS) [@webb2011statistical] algorithm can help determine an appropriate normalization approach. This algorithm ranks different combinations of subset and normalization functions based on a score that captures how much bias a particular normalization procedure introduces into the data. Higher scores imply less bias.
+
+```{r spans, eval = FALSE}
+# returns a data frame arranged by descending SPANS score
+# not run due to long runtime; SPANS plot generated and saved to include in vignette
+spans_result <- spans_procedure(mypep)
+
+plot(spans_result)
+```
+
+
+```{r spans plot, out.width = "600px", echo=FALSE}
+knitr::include_graphics("SPANS_pep.png")
+```
+
+
+```{r}
+head(spans_result)
```

Comment:
`spans_result` doesn't exist during build, I think we needed to load `spans_result.RDS` before here.  The committed spans_result.RDS didn't match the results shown in SPANS_pep.png but re-running SPANS gives the same result as the .png - I have an updated spans_result.RDS in the github actions pull request.

Reply:
Ah I see, the spans_result.RDS looks like the one for isobaric pepdata.

Reply:
Didn't do anything here...seemed resolved?

Reply:
sry didn't mean to imply this was resolved, see my comment in the new PR